modules/oeFax/_credentials.php

```diff
@@ -0,0 +1,15 @@
+<?php
+return array(
+    'username'      => '',
+    'extension'     => '',
+    'password'      => '',
+    'appKey'        => '',
+    'appSecret'     => '',
+    'server'        => '',
+    'portal'        => '',
+    'smsNumber'     => '',
+    'production'    => '',
+    'redirect_url'  => '',
+    'smsHours'      => "",
+    'smsMessage'    => "A courtesy reminder for ***NAME*** \r\nFor the appointment scheduled on: ***DATE*** At: ***STARTTIME*** Until: ***ENDTIME*** \r\nWith: ***PROVIDER*** Of: ***ORG***\r\nPlease call if unable to attend.",
+);
```

Comment:
Are we storing credentials in the codebase. Note this will break on multisite and doesn't align with futuristic goal of making OpenEMR stateless (ie. all stuff stored in database(mysql/mariadb/couchdb)). Agree we should not be using globals for all these. Why not create a mysql table to store this?

Reply:
just a template. never populates

Reply:
> Why not create a mysql table to store this?

Because sooner or later we'll have to deal with file configurable setup and/or configs. Plus I don't want another tiny table for a purpose that 80% won't use.

> Are we storing credentials/settings in the codebase.

No, in sites and multisite is covered.

Reply:
Why would we go back to file configs (in the far distant past all the "globals" were in a file in addition to another config file and it was a complete mess)?
Why not just make a flexible module_settings table with 2 sql columns, the `id` (which would be oefax for this entry) and `settings` which is simply the json of settings in a sql text column. Then have a flexible place for all module settings (1 sql table for them all) and it won't drag on the globals. Also note the long term evil goal is to make the openemr codebase as stateless as possible to optimize it for swarm/kubernetes use (so if you do place the settings in a file then some poor developer will likely need to migrate it to the database in the future :) ).